wgsl/index.bs

```diff
@@ -3959,6 +3959,278 @@ TODO: deduplicate these tables
+  <tr><td> `texture_sampled_2d_array<type>`   <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_sampled_3d<type>`         <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_sampled_cube<type>`       <td>        <td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_sampled_cube_array<type>` <td>        <td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_ro_1d<type>`              <td>&#10003;<td>        <td>        <td>
+  <tr><td> `texture_ro_1d_array<type>`        <td>&#10003;<td>        <td>        <td>
+  <tr><td> `texture_ro_2d<type>`              <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_ro_2d_array<type>`        <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_ro_3d<type>`              <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_depth_2d`                 <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_depth_2d_array`           <td>&#10003;<td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_depth_cube`               <td>        <td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `texture_depth_cube_array`         <td>        <td>&#10003;<td>&#10003;<td>&#10003;
+</table>
+
+<table class='data'>
+  <caption>Compatibility with other named arguments</caption>
+  <thead>
+    <tr><td><td>`offset`<td>`bias` <td>`lod`  <td>`grad`
+  </thead>
+  <tr><td> `offset` <td>        <td>&#10003;<td>&#10003;<td>&#10003;
+  <tr><td> `bias`   <td>&#10003;<td>        <td>        <td>
+  <tr><td> `lod`    <td>&#10003;<td>        <td>        <td>
+  <tr><td> `grad`   <td>&#10003;<td>        <td>        <td>
+</table>
+
+### Sample & Compare ### {#texture-sample-compare-functions}
+
+Sampling from a depth texture with a depth-comparison uses the `textureSampleCompare` function:
+
+```
+f32 textureSampleCompare(<depth_texture>, <sampler_comparison>, <coordinates>, f32 depth_reference [, offset: <offset>])
+```
+
+Where:
+
+* `<depth_texture>` is a [depth](#texture-depth) texture type.
+
+* `<sampler_comparison>` is a [sampler comparision](#sampler-type) type.
+
+* `<coordinates>` are the texture coordinates used for sampling.
+    The type for the texture coordinates depends on the `<texture>` type:
+
+    <table class='data'>
+      <thead>
+        <tr><td>`<texture>` type<td>`<coordinates>` type
+      </thead>
+      <tr><td> `texture_depth_2d`         <td> `vec2<f32>`
+      <tr><td> `texture_depth_2d_array`   <td> `vec3<f32>`
+      <tr><td> `texture_depth_cube`       <td> `vec3<f32>`
+      <tr><td> `texture_depth_cube_array` <td> `vec4<f32>`
+    </table>
+
+    Array textures (with `_array` suffix) use the last component of `<coordinates>`
+    to specify array index.
+
+
+Samples the depth value at `<coordinates>`, and compares the sampled value against `depth_reference`
+and returns an `f32` in the range `[0.0 - 1.0]`.
```

Comment:
nit: perhaps `[0.0 .. 1.0]` would be cleaner?

Reply:
Done